code/idb/idbstandard.q

```diff
@@ -0,0 +1,15 @@
+
+\d .idb
+hdbdir:hsym @[value;`hdbdir;`:hdb];                                        /-set hdb directory. Will be used for reading symbols from sym file
+savedir:hsym @[value;`savedir;`:wdb];                                      /-set wdb directory.
+partitiontype:@[value;`partitiontype;`date];                               /-set type of partition (defaults to `date)
+getpartition:@[value;`getpartition;                                        /-function to determine the partition value
+               {{@[value;`.idb.initialpartition;
+                   (`date^partitiontype)$.proc.cd[]]}}];
+
+initialpartition:getpartition[];                                           /-at startup the IDB will try to load this partition. Later on the WDB instructs the IDB which partition to load.
+setupsuccess:0b;                                                           /-the IDB is only considered to be successfully setup if it was able to load the underlying database(there is data and sym file)
+
+\d .proc
+trap:1b;                                                                   /-the IDB won't fail, it just signals error on its stderr. This means that the process will be still alive
+                                                                           /-when there is no database behind. This will prevent IDB failures during EOD merging of the WDB.
```

Comment:
do we want this? I see from comment it's to allow for case where there is no database - can we not just detect that & not error in that case? this changes the standard behaviour of TorQ processes which is to fail fast if there are any errors during start up (which could be any number of things besides there being no database to load)

Reply:
I now changed that completely. The IDB will register itself with the WDB and retrieve the parameters needed. Also the WDB will initialise an empty db each time after EOD.